### ISSUE TITLE
Sentinel Password

### ISSUE THREAD

**Issue by nivaldodasilva** [luin/ioredis#318]
For some reason `redis-sentinel.conf` is requesting for password so either it should have a password parameter on the endpoint or the `redis-sentinel.conf` should set `protected-mode no`.

For document purpose, this is error I get when I should pass the password to the sentinel connection but I dont:

```
[ioredis] Unhandled error event: Error: read ECONNRESET
    at exports._errnoException (util.js:870:11)
    at TCP.onread (net.js:552:26)
```

```
SentinelConnector.prototype.resolve = function (endpoint, callback) {
    if (typeof Redis === 'undefined') {
        Redis = require('../redis');
     }
     var client = new Redis({
     port: endpoint.port,
     host: endpoint.host,
     ***password: endpoint.password,***
     retryStrategy: null,
     enableReadyCheck: false,
     connectTimeout: this.options.connectTimeout,    
     dropBufferSupport: true
});
...
```

**Comment by luin** [comment]
As far as I know, sentinel doesn't support setting a password (more precisely, allow setting a password with `requirepass` directive but not support `AUTH` command).

For the sentinel bundled with Redis 3.2, which has protected-mode enabled, you should use the `bind` directive to restrict Redis to certain interfaces.

BTW, when denied by the protect mode, ioredis should return a more informative error message, here're what I got:

```
[ioredis] Unhandled error event: Error: read ECONNRESET
    at exports._errnoException (util.js:870:11)
    at TCP.onread (net.js:552:26)
[ioredis] Unhandled error event: Error: write EPIPE
    at exports._errnoException (util.js:870:11)
    at WriteWrap.afterWrite (net.js:769:14)
[ioredis] Unhandled error event: Error: Connection is closed.
    at close (/Users/luin/opensource/ioredis/lib/redis/event_handler.js:99:21)
    at Socket.<anonymous> (/Users/luin/opensource/ioredis/lib/redis/event_handler.js:79:14)
    at Socket.g (events.js:260:16)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at TCP._onclose (net.js:477:12)
[ioredis] Unhandled error event: ReplyError: DENIED Redis is running in protected mode because protected mode is enabled, no bind address was specified, no authentication password is requested to clients. In this mode connections are only accepted from the loopback interface. If you want to connect from external computers to Redis you may adopt one of the following solutions: 1) Just disable protected mode sending the command 'CONFIG SET protected-mode no' from the loopback interface by connecting to Redis from the same host the server is running, however MAKE SURE Redis is not publicly accessible from internet if you do so. Use CONFIG REWRITE to make this change permanent. 2) Alternatively you can just disable the protected mode by editing the Redis configuration file, and setting the protected mode option to 'no', and then restarting the server. 3) If you started the server manually just for testing, restart it with the '--protected-mode no' option. 4) Setup a bind address or an authentication password. NOTE: You only need to do one of the above things in order for the server to start accepting connections from the outside.
```

**Comment by scality-gdoumergue** [comment]
Hello,

I would like to re-consider authentication upon redis-sentinel, because it's now available thanks to [this commit](https://github.com/antirez/redis/commit/bdf6306f2979adbe009805959ef05efb5da80ff6), available since redis 5.0.

Thanks a lot

**Comment by luin** [comment]
> Hello,
> 
> I would like to re-consider authentication upon redis-sentinel, because it's now available thanks to [this commit](https://github.com/antirez/redis/commit/bdf6306f2979adbe009805959ef05efb5da80ff6), available since redis 5.0.
> 
> Thanks a lot

Authentication for sentinel has already been supported in ioredis v4.9.0: https://github.com/luin/ioredis/blob/master/Changelog.md#490-2019-03-18
